Made ListBox.DefaultItemHeight property obsolete.

dkazennov opened the pull request:
Fixes #4463 (partially, see also #8755)

## Proposed changes

- The ListBox.DefaultItemHeight constant marked as obsolete since the `ListBox.ItemHeight` after #8518 will be depended on actual `DefaultFont.Height` value.
- New unit test `ListBox_Height_IsEqualDefaultHeight_IfDefaultFontIsChanged` is added.
- Take note: it should be merged after the #8755 

<!-- We are in TELL-MODE the following section must be completed -->

## Customer Impact

- Made property obsolete. Made changes to documentation and obsoletion inside the project.

## Regression? 

- No

## Risk

- Minimal

<!-- end TELL-MODE -->


## Test methodology

- Manual (Issue can be tested via substitution of dlls, complete restart of the VS recommended)

## Test environment(s) <!-- Remove any that don't apply -->

- dotnet 8.0.100-alpha.1.22607.6


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8518)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8640)

dkazennov commented:
Note: This PR should be merged after #8518.

Tanya-Solyanik commented:
Please move the corresponding test changes as well. After the attribute removal change is in, we'll expose the replacement property for this constant.

Tanya-Solyanik commented:
Related to https://github.com/dotnet/docs/issues/33953

dkazennov commented:
Rebased to incorporate changes, introduced by #8518
Now new unit test passes successfully.

dkazennov commented:
Rebased on most recent main to include massive fixes in unit tests. Changed the unit test `ListBox_ItemHeight_EqualsDefaultHeight_IfDefaultFontIsChanged` to be in line with fix #8755

dkazennov commented:
I have changed `ListBox_ItemHeight_EqualsDefaultHeight_IfDefaultFontIsChanged` to be in line with changes in #8755 (investigation there showed issues in this PR as well). Of course, this PR will work only after #8755 is merged.

Take a note this new test here `ListBox_ItemHeight_EqualsDefaultHeight_IfDefaultFontIsChanged` now intersects a bit with new test `ListBox_ItemHeight_ResetValue_Success_If_DefaultListBoxItemHeight_IsChanged` introduced in #8755.

lonitra commented:
Closing as the approach to fix this issue seems to be superseded by https://github.com/dotnet/winforms/pull/10142
